framework/src/st_things/things_stack/src/common/framework/things_data_manager.c

```diff
@@ -0,0 +1,2271 @@
+		return false;
+	}
+	// Set Main-Device ID
+	id = OCGetServerInstanceIDString();
+	dev_list[0] = (st_device_s *) hashmap_get(g_device_hmap, (unsigned long)0);
+
+	if (id == NULL || dev_list[0] == NULL) {
+		THINGS_LOG_V_ERROR(THINGS_ERROR, TAG, "Error : id = %s, Main-device =0x%X", id, dev_list[0]);
+		goto GOTO_ERROR;
+	}
+	memcpy(dev_list[0]->device_id, id, strlen(id) + 1);
+
+	// Set Sub-Device ID
+	for (i = 1; i < device_cnt; i++) {
+		dev_list[i] = (st_device_s *) hashmap_get(g_device_hmap, (unsigned long)i);
+		if (dev_list[i] == NULL) {
+			THINGS_LOG_V_ERROR(THINGS_ERROR, TAG, "Error : device[%d] =0x%X", i, dev_list[i]);
+			goto GOTO_ERROR;
+		}
+
+		if (user_def_dev_list[i] != NULL || strlen(dev_list[i]->device_id) == 0) {
+			// a. If logical device is in a hosting(physical) device.
+			if (dev_list[i]->is_physical == 1) {
+				OCRandomUuidResult idGenRet = RAND_UUID_CONVERT_ERROR;
+
+				if (user_def_dev_list[i]) {
+					THINGS_LOG_D(THINGS_DEBUG, TAG, "Set Sub-device ID with user-defined device ID.");
+					memcpy(dev_list[i]->device_id, user_def_dev_list[i], strlen(user_def_dev_list[i]) + 1);
+					idGenRet = RAND_UUID_OK;
+				} else {
+					THINGS_LOG_D(THINGS_DEBUG, TAG, "Set Sub-device ID with Auto-Generated device ID.");
+					idGenRet = OCGenerateUuidString(dev_list[i]->device_id);
+				}
+
+				if (idGenRet != RAND_UUID_OK) {
+					THINGS_LOG_D_ERROR(THINGS_ERROR, TAG, "[DEVICE] Device ID Creation Failed : %d", idGenRet);
+				} else {
+					THINGS_LOG_D(THINGS_DEBUG, TAG, "[DEVICE] ID Created : %s", (dev_list[i]->device_id));
+				}
+			}
+			// b. If logical device is in the other physical device
+			else {
+				THINGS_LOG_D(THINGS_INFO, TAG, "[DEVICE] Not Physically Separated Device ");
+			}
+		}
+	}
+
+	is_support_user_def_dev_list = false;	//  It's allow permission to apply user defined device-ID only when start THINGS_STACK.
+
+	dm_delete_user_define_device_id();
+	things_free(dev_list);
+	dev_list = NULL;
+	return true;
+
+GOTO_ERROR:
+	things_free(dev_list);
+	dev_list = NULL;
+	return false;
+}
+
+int dm_register_resource(things_server_builder_s *p_builder)
+{
+	THINGS_LOG_D(THINGS_DEBUG, TAG, THINGS_FUNC_ENTRY);
+
+	if (p_builder == NULL) {
+		THINGS_LOG_ERROR(THINGS_ERROR, TAG, "Builder instance is NULL");
+
+		return 0;
+	}
+
+	int device_cnt = (int)hashmap_count(g_device_hmap);
+	if (device_cnt < 1) {
+		THINGS_LOG_ERROR(THINGS_ERROR, TAG, "No Device Information to Register");
+
+		return 0;
+	}
+
+	for (int device_num = 0; device_num < device_cnt; device_num++) {
+		char id[11] = { 0 };
+		char res_uri[MAX_URI_LENGTH] = { 0 };
+		int key = 0;
+		int n_count_of_children = 0;
+
+		st_device_s *device = NULL;
+		struct things_resource_info_s *resource = NULL;
+		struct col_resource_s *col_resource_s = NULL;
+		struct things_resource_s *p_collection_resource = NULL;
+
+		// 2. Device Capability Resources Registration
+		THINGS_LOG_D(THINGS_DEBUG, TAG, "BEFORE SEARCHING THE DEVICE 
```

Comment:
Please, remove line 2263 ~ 2264.

Reply:
Because files have changed so many times, could you mention the specific sentences? I cannot search that.

Reply:
2260 	return Parse_things_files(devJsonPath);
2261 }
2262 
2263 //
2264 
2265 int dm_termiate_module()

Reply:
removed